lib/config/utils.ts

```diff
@@ -19,7 +19,8 @@ export function mergeChildConfig<
     if (
       option.mergeable &&
       childConfig[option.name] &&
-      parentConfig[option.name]
+      parentConfig[option.name] &&
+      parentConfig[option.name] !== childConfig[option.name]
```

Comment:
PLease do this in separate PR and add a testcase for it

Reply:
Is this still required after #18858?

Reply:
no, it is not required anymore. i removed it